### DISCUSSION TITLE
How to swap out specific terms within a locale for different types of users

### DISCUSSION THREAD

**Discussion by JakeBriscoe** [ivanhofer/typesafe-i18n#716] [Q&A] [answered]
I have a set of phrases in an `en` translation file. For specific key terms I want to localise these too based on the type of the user. As an example a user could own a fruit store or a bakery, I want to do something like the below but I'm not sure on the best way to do this. 

```
const en = {
	title: 'Manage your {fruit store || bakery} here',
	product: 'Purchase more {fruit || baking ingredients}?',
} satisfies BaseTranslation;
```

I could create additional translation files but then the locale would be something like `en-fruit` and I figured this wasn't recommended. 

I've got it half working using a formatter. But I got an error when trying to read the userType from the page data so I've hard coded the user type in the formatter. And I'm having to pass in the term I want to change when calling the translation, but I want the translation file to already have the term and pass it directly to the formatter. 

```
// formatter
term: (term: Terms) => termsMapping[userType]?.[term]

// translation
title: 'Manage your {0:Terms|term} here',

// +page.svelte
<h2>{$LL.home.title('fruit')}</h2>
```

I hope that made sense 😊

**Comment by ivanhofer** [comment, marked as answer]
From what I understand formatters are the thing you are looking for. But indeed, you don't have access to translations in there.
You could create a utility function to get to the translated value instead of using a formatter.

```ts
// util
const toTerm = (term: Terms) => $LL.terms[term]()

// translation
title: 'Manage your {0:LocalizedString} here',
terms: {
   fruit: 'fruit store',
   bakery: 'bakery',
}

// +page.svelte
<h2>{$LL.home.title(toTerm(user.type))}</h2>
```

Hope this is clear
